pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter_test.go

```diff
@@ -267,3 +223,20 @@ var _ = Describe("Validating MigrationCreate Admitter", func() {
 		)
 	})
 })
+
+func newAdmissionReviewForVMIMCreation(migration *v1.VirtualMachineInstanceMigration) (*admissionv1.AdmissionReview, error) {
+	migrationBytes, err := json.Marshal(migration)
+	if err != nil {
+		return nil, err
+	}
+
+	return &admissionv1.AdmissionReview{
+		Request: &admissionv1.AdmissionRequest{
+			Resource: webhooks.MigrationGroupVersionResource,
+			Object: runtime.RawExtension{
+				Raw: migrationBytes,
+			},
+			Operation: admissionv1.Create,
+		},
+	}, nil
+}
```

Comment:
You've referenced a [similar function](https://github.com/kubevirt/kubevirt/blame/main/pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go#L75) for migration-create-mutator_test.go in the commit message. You also state that this is a follow-up to the PR that have introduced this function.

Instead of introducing many more PRs that will do the same, perhaps we can avoid this code duplication and create a single function to serve all, that can get as a parameter the groupVersionKind of the relevant resource?

Reply:
I agree that a generic function can be created for that purpose, I'll check what can be done.
@orelmisan WDYT?

Reply:
I think that this is a fine optimization that can be considered, but the goal is to get rid of gomock and make use of the generated clientset. Perhaps we can consolidate that logic at a later time once we complete the directive we've opted for in the issue, WDYT @iholder101?

Reply:
@iholder101 Similar functions exist for the other webhooks' tests that were recently refactored:
https://github.com/kubevirt/kubevirt/blob/4fdbdbfe85e565862e6d6a13807e2e9d5de68dfd/pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator_test.go#L155

https://github.com/kubevirt/kubevirt/blob/4fdbdbfe85e565862e6d6a13807e2e9d5de68dfd/pkg/virt-api/webhooks/mutating-webhook/mutators/migration-create-mutator_test.go#L75

https://github.com/kubevirt/kubevirt/blob/4fdbdbfe85e565862e6d6a13807e2e9d5de68dfd/pkg/virt-api/webhooks/validating-webhook/admitters/migration-update-admitter_test.go#L232

https://github.com/kubevirt/kubevirt/blob/4fdbdbfe85e565862e6d6a13807e2e9d5de68dfd/pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go#L498

The advantage of having these functions per object and operation is the fact that it is easy to understand what their return value will be.
I personally think that trying to generalize them into a single function may reduce code readability, but nevertheless I'm in favor of exploring this direction in a follow-up.

Reply:
After looking into it, I believe the effort in trying to create a generic function that handles all admission review creation would be more detrimental to code quality rather than beneficial (that's without taking into account the relatively complex logic such function would require in its implementation). Those functions, despite having near identical logic, serve different purposes, thus they don't violate the DRY principle.

Reply:
@Acedus @orelmisan Hey folks. I think that the concern is real. I think that if we keep this PR as is, we will add another function to the code that we would later need to refactor again. The current state of this PR is that it's easy to understand, and I don't think that if you will add a generic package to create admission review package it will make this PR complicated. Vice versa, we can stop that train of PRs with duplicated functions now and provide a solution here. I will review it

Reply:
Hi @enp0s3, that's fine by me. I think it would better fit a separate PR so I'll do that.
My idea was something akin to [libvmi](https://github.com/kubevirt/kubevirt/tree/main/pkg/libvmi), so something along the lines of:
```go
package libadmission

import (
	"encoding/json"

	admissionv1 "k8s.io/api/admission/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/apimachinery/pkg/runtime"
)

// Option represents an action that enables an option.
type Option func(ar *admissionv1.AdmissionReview)

// New instantiates a new AdmissionReview configuration,
// building its properties based on the specified With* options.
func NewAdmissionReview[T any](object *T, opts ...Option) (*admissionv1.AdmissionReview, error) {
	ar, err := baseAdmissionReview(object)
	if err != nil {
		return nil, err
	}

	for _, f := range opts {
		f(ar)
	}

	return ar, nil
}

func WithOperation(operation admissionv1.Operation) Option {
	return func(ar *admissionv1.AdmissionReview) {
		ar.Request.Operation = operation 
	}
}

func baseAdmissionReview[T any](object *T) (*admissionv1.AdmissionReview, error) {
	objectBytes, err := json.Marshal(object)
	if err != nil {
		return nil, err
	}

	return &admissionv1.AdmissionReview{
		Request: &admissionv1.AdmissionRequest{
			Object: runtime.RawExtension{
				Raw: objectBytes, 
			},
		},
	}, nil
}
...
```

WDYT?

Reply:
@Acedus Hi, yes exactly! IMO it would be great to have this approach.

Reply:
@Acedus Although I had a conversation with @orelmisan yesterday and seems that it could be difficult to make it generic, for example there is a need to mention the CRD name and version. But if you will make it work it could be great!

Reply:
@Acedus @enp0s3, this approach will not improve code readability IMO.
Imagine having a call with a complex builder, one could argue that it is too complex, wrap it in a function to simplify it - and then we are back to square one but with an unreadable function.

Is it possible to move forward with this PR and explore additional options in a follow-up?

Reply:
@orelmisan SGTM

Reply:
```go
actualResponse := admitter.Admit(newAddmissionReview(WithOption1(), WithOption2(),WithOption3()))
```

One will probably want to warp it:
```go
func newAdmissionReviewForVMIMCreation(vmim *VirtualMachineInterfaceMigration) *admissionv1.AdmissionReview {
    return newAddmissionReview(WithOption1(), WithOption2(),WithOption3()
}
```

Reply:
I agree with @orelmisan that it would simply result in more wrapping despite the possible extensibility of it, one gripe I had with it personally is that unlike libvmi it doesn't really have anything to do with KubeVirt. 

As for GVR/GVK, this is just a general question (perhaps I'm missing something) but can't we rely on the fact that all tested objects satisfy the `ObjectKind` interface and have the `GroupVersionKind` method?

In any event, I'm indifferent and wouldn't mind circling back to the implementation of a generic way to construct admission reviews later should we need it. Thanks for the thorough review!